app/code/Magento/NewsletterApi/Api/Data/SubscriptionInterface.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\NewsletterApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+use Magento\NewsletterApi\Api\Data\SubscriptionExtensionInterface;
+
+/**
+ * Newsletter Subscription Interface
+ *
+ * represents one newsletter subscription identified by an
+ * e-mail address or the internal id
+ *
+ * @api
+ */
+interface SubscriptionInterface extends ExtensibleDataInterface
+{
+    /**#@+
+     * Constants for keys of data array. Identical to the name of the getter in snake case
+     */
+    const SUBSCRIPTION_ID = 'subscription_id';
+    const EMAIL = 'email';
+    const STATUS = 'status';
+    /**#@-*/
+
+    /**
+     * Set ID
+     *
+     * set the internal id of the entity
+     *
+     * @param int $id
+     * @return void
+     */
+    public function setId($id);
+
+    /**
+     * Set E-Mail Address
+     *
+     * set the e-mail address of the subscriber
+     *
+     * @param string $email
+     * @return void
+     */
+    public function setEmail(string $email);
+
+    /**
+     * Set Status
+     *
+     * set the status of the subscription
+     *
+     * @param int $status
```

Comment:
you cannot predefine the status in the api since different implementations might have different status. Int was just a decision because the current Magento_Newsletter implementation uses integer constants